plus/api/src/main/java/org/sakaiproject/plus/api/model/Context.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2021- Charles R. Severance
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.sakaiproject.plus.api.model;
+
+import javax.persistence.Column;
+import javax.persistence.Lob;
+import javax.persistence.Embedded;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.Index;
+import javax.persistence.UniqueConstraint;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import javax.persistence.CascadeType;
+
+import org.hibernate.annotations.GenericGenerator;
+
+import org.sakaiproject.springframework.data.PersistableEntity;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Table(name = "PLUS_CONTEXT",
+  indexes = { @Index(columnList = "CONTEXT, TENNANT_GUID") },
+  uniqueConstraints = { @UniqueConstraint(columnNames = { "CONTEXT", "TENNANT_GUID" }) }
+)
+@Getter
+@Setter
+public class Context extends BaseLTI implements PersistableEntity<String> {
+
+    @Id
+    @Column(name = "CONTEXT_GUID", length = 36, nullable = false)
+    @GeneratedValue(generator = "uuid")
+    @GenericGenerator(name = "uuid", strategy = "uuid2")
+    private String id;
+
+    @Column(name = "CONTEXT", length = 1024, nullable = false)
+    private String context;
+
+	@ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
```

Comment:
Is this the behaviour you wanted? If you delete a Context via your repository, this will delete the associated Tenant record, thus orphaning any other Context records that had that Tenant as their association (unless you have a foreign key  relationship defined). If your intention is to persist a Tenant when you persist a Context, then you could use CascadeType.PERSIST. Then (I think) you could create a Context and a new Tenant together and just save the Context via your repository. Hibernate will then save the Tenant too.

Reply:
Woah - That is not what I wanted.  What I was hoping for was to make it so when a Tenant was deleted, all the associated contexts would be deleted.